pandas/core/generic.py

```diff
@@ -671,14 +657,7 @@ def set_axis(self, labels, axis=0, inplace=None):
             labels, axis = axis, labels
 
         if inplace is None:
-            warnings.warn(
-                "set_axis currently defaults to operating inplace.\nThis "
-                "will change in a future version of pandas, use "
-                "inplace=True to avoid this warning.",
-                FutureWarning,
-                stacklevel=2,
-            )
-            inplace = True
+            raise ValueError("`inplace` cannot have None as a value.")
```

Comment:
I think you can just delete this altogether - don't think we do this anywhere else (though correct if wrong)

Reply:
No, I think you're right